doc/source/index.rst

```diff
@@ -9,84 +9,179 @@ Ray
-|  # Execute f serially.                         |  # Execute f in parallel.                          |
-|                                                |                                                    |
-|                                                |  @ray.remote                                       |
-|  def f():                                      |  def f():                                          |
-|      time.sleep(1)                             |      time.sleep(1)                                 |
-|      return 1                                  |      return 1                                      |
-|                                                |                                                    |
-|                                                |                                                    |
-|                                                |  ray.init()                                        |
-|  results = [f() for i in range(4)]             |  results = ray.get([f.remote() for i in range(4)]) |
-+------------------------------------------------+----------------------------------------------------+
-
-To launch a Ray cluster, either privately, on AWS, or on GCP, `follow these instructions <autoscaling.html>`_.
+Install Ray with: ``pip install ray``. For nightly wheels, see the `Installation page <installation.html>`__.
 
 View the `codebase on GitHub`_.
 
 .. _`codebase on GitHub`: https://github.com/ray-project/ray
 
-Ray comes with libraries that accelerate deep learning and reinforcement learning development:
 
-- `Tune`_: Scalable Hyperparameter Search
-- `RLlib`_: Scalable Reinforcement Learning
-- `Distributed Training <distributed_training.html>`__
+Quick Start
+-----------
+
+.. code-block:: python
+
+    ray.init()
+
+    @ray.remote
+    def f(x):
+        return x * x
+
+    futures = [f.remote(i) for i in range(4)]
+    print(ray.get(futures))
+
+To use Ray's actor model:
+
+.. code-block:: python
+
+    ray.init()
+
+    @ray.remote
+    class Counter():
+        def __init__(self):
+            self.n = 0
+
+        def inc(self):
+            self.n += 1
+
+        def read(self):
+            return self.n
+
+    counters = [Counter.remote() for i in range(4)]
+    [c.increment.remote() for c in counters]
+    futures = [c.read.remote() for c in counters]
+    print(ray.get(futures))
+
+
+Ray programs can run on a single machine, and can also seamlessly scale to large clusters. To execute the above Ray script in the cloud, just download `this configuration file <https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/aws/example-full.yaml>`__, and run:
+
+``ray submit [CLUSTER.YAML] example.py --start``
+
+See more details in the `Cluster Launch page <autoscaling.html>`_.
+
+Tune Quick Start
+----------------
+
+`Tune`_ is a scalable framework for hyperparameter search with a focus on deep learning and deep reinforcement learning.
+
+.. code-block:: python
```

Comment:
This is a great example, but I am concerned that a lot of people won't have pytorch installed and won't be able to run this.

I think this can be addressed by just adding

```python
# Use 'pip install torch torchvision' to install PyTorch.
```

as the first line of the script (should work on Mac/Linux, and they can ignore if they already have it installed).

Alternatively, we could use something simpler here, e.g., scikitlearn. Though maybe scikitlearn is just as hard to install..